### ISSUE TITLE
Products assigned to mutiple categories but cannot show on some product category pages

### ISSUE THREAD

**Issue by wisenet** [superspeedyplugins/issue-tracker#126]
**Describe the bug**
I have some products that assigned in multiple categories, e.g. some product has been assigned in both
Performance>Braking System>Brake Pads and Shoes
Regular Services>Brake Pads and Shoes

I have checked that the slug has no conflict and both sub-categories "Brake Pads and Shoes" has two different filter set with same setting and filters. However don't know why product only shows in "Regualr services" but there no product shows in "Performance>Braking System>Brake Pads and Shoes"
And I found that the Breadcrumb and URL string works correct in "Regular Service>Brake Pads and Shoes"
https://uat.sgt-store.com/category/services/regular-services/brake-pads-and-shoes-regular-services/?stock-status=show-all
BUT not correct on "Performance>Baking System>Brake Pads and Shoes", the Breadcrubm and URL jumps and look like
Breadcrumb should be "HOME / SHOP / PERFORMANCE / BRAKE PADS AND SHOES" BUT it shows "HOME / SHOP / PERFORMANCE / PERFORMANCE / BRAKE PADS AND SHOES / PERFORMANCE"
and the URL: https://uat.sgt-store.com/category/performance/braking-system/brake-pads-and-shoes-performance/performance/?stock-status=show-all

Please reference to the screenshot video
https://drive.google.com/file/d/1BMJUftWIEGrFl5H-78n7l11MySK_8KK3/view?usp=drive_link

I have disabled all others plugins to test but same result and I then disabled SSF plugin and test again with above categories, all works and product shows correctly

**To Reproduce**
1. Go to 'https://uat.sgt-store.com/category/services/regular-services/brake-pads-and-shoes-regular-services/'
2. Products shows correct in sub category 'Services>Regular Services>Brake Pads and Shoes'
3. Products assigned to both 'Services>Regular Services>Brake Pads and Shoes" and "Performance>Braking System>Brake Pads and Shoes'
4. Then go to and compare 'https://uat.sgt-store.com/category/performance/braking-system/brake-pads-and-shoes-performance/'
5. Click 'Show All' on SSF at left sidebar
6. No products shows at this product category but the products has been assigned same as in 'Services>Regular Services>Brake Pads and Shoes'

**Expected behavior**
Products shows same as in 'Services>Regular Services>Brake Pads and Shoes'

**Screenshots**
https://drive.google.com/file/d/1Mkhr1jDkysmuuCMQQ4DhOQORmxQeSKd2/view?usp=sharing
https://drive.google.com/file/d/1Mloq7tthip4Ae40kkbONtNYsfaF65zzM/view?usp=drive_link
https://drive.google.com/file/d/1MrjpFp5qeKifdlU3uL3zZBshcNDYthL5/view?usp=drive_link
https://drive.google.com/file/d/1MsG29sx-Z-vSdXyUVbNNEgGnUUmnA1NX/view?usp=drive_link
screenshot video
https://drive.google.com/file/d/1BMJUftWIEGrFl5H-78n7l11MySK_8KK3/view?usp=drive_link

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version: 123.0.6312.122 (Official Build) (64-bit)

**Smartphone (please complete the following information):**

**Additional context**
I have tried 
- disabled all other plugins
- clear server and browser cache
- updated SSF to latest version
- Swtich to different theme
- Disable CDN
but still not fixed the issue

**Comment by dhilditch** [comment]
Tested this on my local, created two hierarchical categories:

Main One > Main Two
Secondary One > Secondary Two

Assigned a couple of products to Main Two and Secondary Two.

Main Two is working: https://i.imgur.com/iiRajDJ.png
Secondary Two is also working: https://i.imgur.com/2e9QnVy.png

I wonder if maybe the issue is coming from another bug in our table maintenance code. @wisenet could you confirm if you copy/paste the SQL from the Settings > SSF > Advanced tab and run it manually then flush your object cache, does it work afterwards?

**Comment by wisenet** [comment]
Hi @dhilditch,

have you try 
Cat A>sub cat A01
Cat B>Sub Cat B01>Sub Cat B002

I don't use any permalink plugin, permalink setting in below screenshot
https://drive.google.com/file/d/1MwQSIALvK_gUiO2rHfUwjPJyHu7QkYXy/view?usp=drive_link
https://drive.google.com/file/d/1MzHMFaAJWmsvoinkZAh6vPStLdNNbw3L/view?usp=drive_link

I'm running OpenLiteSpeed Server and PHP8.2, no cache plugin

Thanks

**Comment by dhilditch** [comment]
Yes, tried - i created all the above just now, I tried adding them to ALL categories and just to individual ones from above.

Can you confirm you are on the SSF latest version?

There was something which might have caused issues in an earlier version pre 5.44 if you performed updates to products.

https://www.superspeedyplugins.com/support/changelogs/super-speedy-filters/

Can you install the very latest 5.45, do a full table rebuild and then retest this?

Failing that, I'll need to get access to your dev site and SFTP etc so I can add some debugging and figure out whats' happening.

**Comment by wisenet** [comment]
Hi @dhilditch,

I have updated SSF latest version 5.46, manual rebuild table, clear and disabled cache plugin and products still cannot show in some product categories

Screen record
https://drive.google.com/file/d/1uFlMeDlBlZ7m9uW1o-usA6VlE3raKG0l/view?usp=drive_link

And when there some warning say missing table when manual rebuild
Note: #1051 Unknown table 'AotSC1Lz0b0SIG.wp_fww_ancestors_temp'
Note: #1051 Unknown table 'AotSC1Lz0b0SIG.wp_fww_ancestors'

FTP:
uat.sgt-store.com
Port: 21
Login: admin_davehilitch
password: 7ssLSix%)^r3awZ(Byj&RmQI

**Comment by dhilditch** [comment]
I could not reproduce this issue. Can you try and reproduce this bug? @wisenet you shouldn't leave your FTP details in here, you should message them directly to us - in this case @hassan_18193 on discord @Hassan Azhar will be taking over development of the fix for this.

I could not reproduce, so he will need access to your environment to figure out the cause so we can figure out the solution.

**Comment by dhilditch** [comment]
Checked against SSF 6.40.4. Product-category membership is still resolved through the main filter clauses and object_ancestors data, so a stale or incomplete ancestor row can exclude products assigned to multiple categories. Suggested fix: add a multi-category regression fixture and verify rebuild/update paths preserve every term relationship.

**Comment by dhilditch** [comment]
Tested current core SSF 6.54.1 (`beca1c068b2117d974df99e27a4667ceb9e048f4`) on its own development site.

Recreated both Services/Regular Services and Performance/Braking System branches with a shared Brake Pads and Shoes product. Turning the stock checkbox off returned both in-stock and out-of-stock products without duplicating breadcrumb ancestors or appending taxonomy parents to the URL. Passed with URL rewriting disabled and enabled.

Dev environment: http://tests-issue-126.super-speedy-filters.localhost:8081/ (local test address).


Closing because the reported defect no longer reproduces in this configuration. The test site is being deleted at Dave's request after preserving this screenshot and the local reproduction records. The environment URL above is a historical reference and will no longer be available.

![Verified working test result](https://github.com/superspeedyplugins/issue-tracker/blob/432bd585f8c52277b9ad4dc72fe2b92bc762b1b1/.github/issue-evidence/ssf-126-20260916.png?raw=true)
